### PR TITLE
Add a datadeffile partition

### PR DESCRIPTION
**Pull request by ikaneshiro**
Includes a partition containing the definition used during the build process for the particular container.

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
given that `SIF` has a `JSON` data partition, I would  have liked to see this PR same functionality on https://github.com/sylabs/singularity/pull/2047 , as I said, I think, the inspect cmd and all that it involves could go into a SIF on a future release.

**Review by ArangoGutierrez — APPROVED** [review]
lgtm 
tested on RHEL7.5

**Review by bauerm97 — DISMISSED** [review]
Blocking merge until @yhcote gets a chance to approve

**Pull request merged**
